core/app/models/spree/store_credit.rb

```diff
@@ -193,13 +219,18 @@ def invalidate(reason, user_performing_invalidation)
       self.update_reason = reason
       self.action_originator = user_performing_invalidation
       self.invalidated_at = Time.current
-      save
+      self.action_amount = ledger_balance
+      debit(ledger_balance, { action_originator: user_performing_invalidation }) if save
```

Comment:
I don't doing work (like save) in a suffix `if` statement. I'd prefer a regular if statement to ensure it is noticed

```
if save
  debit(ledger_balance, { action_originator: user_performing_invalidation })
end
```